stringdtype/stringdtype/src/umath.c

```diff
@@ -125,10 +125,14 @@ string_equal_strided_loop(PyArrayMethod_Context *NPY_UNUSED(context),
     ss *s1 = NULL, *s2 = NULL;
 
     while (N--) {
-        load_string(in1, &s1);
-        load_string(in2, &s2);
-
-        if (s1->len == s2->len && strncmp(s1->buf, s2->buf, s1->len) == 0) {
+        s1 = (ss *)in1;
+        s2 = (ss *)in2;
+        if (ss_isnull(s1) || ss_isnull(s2)) {
```

Comment:
I was thinking that it might be good if `NA` values were considered equal, but it looks like numpy doesn't consider `nan` values equal either, so this seems consistent with that. :+1:

Reply:
Pandas would consider `NA == NA -> NA` but that requires an NA boolean.  I suspect that would be the right thing in principle, though.